internal/app/machined/internal/phase/platform/platform.go

```diff
@@ -25,6 +40,101 @@ func (task *Platform) RuntimeFunc(mode runtime.Mode) phase.RuntimeFunc {
 }
 
 func (task *Platform) runtime(platform platform.Platform, data *userdata.UserData) (err error) {
+	switch platform.Mode() {
+	case runtime.Interactive:
+		var dev *probe.ProbedBlockDevice
+		dev, err = probe.GetDevWithFileSystemLabel(constants.ISOFilesystemLabel)
+		if err != nil {
+			return errors.Errorf("failed to find %s iso: %v", constants.ISOFilesystemLabel, err)
+		}
+
+		if err = unix.Mount(dev.Path, "/tmp", dev.SuperBlock.Type(), unix.MS_RDONLY, ""); err != nil {
+			return err
+		}
+
+		for _, f := range []string{"/tmp/usr/install/vmlinuz", "/tmp/usr/install/initramfs.xz"} {
+			var source []byte
+			source, err = ioutil.ReadFile(f)
+			if err != nil {
+				return err
+			}
+			if err = ioutil.WriteFile("/"+filepath.Base(f), source, 0644); err != nil {
+				return err
+			}
+		}
+
+		reader := bufio.NewReader(os.Stdin)
+		fmt.Print("Talos configuration URL: ")
+		endpoint, err := reader.ReadString('\n')
+		if err != nil {
+			return err
+		}
+
+		cmdline := kernel.NewDefaultCmdline()
+		cmdline.Append("initrd", filepath.Join("/", "default", "initramfs.xz"))
+		cmdline.Append(constants.KernelParamPlatform, "bare-metal")
+		cmdline.Append(constants.KernelParamUserData, endpoint)
+
+		inst := installer.NewInstaller(cmdline, data)
+		if err = inst.Install(); err != nil {
+			return errors.Wrap(err, "failed to install")
+		}
+
+		// nolint: errcheck
+		unix.Reboot(int(unix.LINUX_REBOOT_CMD_RESTART))
+	case runtime.BareMetal:
+		var endpoint *string
+		if endpoint = kernel.ProcCmdline().Get(constants.KernelParamUserData).First(); endpoint == nil {
+			return errors.Errorf("failed to find %s in kernel parameters", constants.KernelParamUserData)
+		}
+
+		cmdline := kernel.NewDefaultCmdline()
+		cmdline.Append("initrd", filepath.Join("/", "default", "initramfs.xz"))
+		cmdline.Append(constants.KernelParamPlatform, "packet")
```

Comment:
i think we'd want to drop this param.

Reply:
or `s/packet/platform.Name()/`

Reply:
Thanks. Yeah this whole bit here needs cleanup. Still a bit of a PoC and hesitant to put more work into if its not a direction we wanna go.